renku/ui/service/serializers/templates.py

```diff
@@ -17,30 +17,51 @@
 
 from urllib.parse import urlparse
 
-from marshmallow import Schema, ValidationError, fields, post_load, pre_load
-from yagup import GitURL
+import yagup
+from marshmallow import Schema, ValidationError, fields, post_load, pre_load, validates
 from yagup.exceptions import InvalidURL
 
 from renku.core.util.os import normalize_to_ascii
 from renku.domain_model.dataset import ImageObjectRequestJson
 from renku.ui.service.config import TEMPLATE_CLONE_DEPTH_DEFAULT
-from renku.ui.service.serializers.cache import ProjectCloneContext, RepositoryCloneRequest
+from renku.ui.service.errors import UserRepoUrlInvalidError
+from renku.ui.service.serializers.common import GitUrlResponseMixin
 from renku.ui.service.serializers.rpc import JsonRPCResponse
 from renku.ui.service.utils import normalize_git_url
 
 
-class ManifestTemplatesRequest(RepositoryCloneRequest):
+class ManifestTemplatesRequest(Schema):
     """Request schema for listing manifest templates."""
 
-    url = fields.String(required=True)
+    template_git_url = fields.String(required=True, metadata={"description": "Template git repository url."})
     depth = fields.Integer(load_default=TEMPLATE_CLONE_DEPTH_DEFAULT)
+    branch = fields.String(load_default=None, metadata={"description": "Remote git branch (or tag or commit SHA)."})
```

Comment:
If I look at the swagger page on dev.renku.ch currently we have `url`, `ref` and `depth` as the parameters for `templates.read_manifest`. Why do we need to switch to branch. I think `ref` is a much more accurate and clearer name for this. Calling it `branch` and using it just like `ref` is really confusing.

Reply:
the swagger gets it from here: https://github.com/SwissDataScienceCenter/renku-python/blob/develop/renku/ui/service/views/templates.py#L56
but the code has always used branch: https://github.com/SwissDataScienceCenter/renku-python/blob/develop/renku/ui/service/controllers/templates_read_manifest.py#L52
which is ultimately defined here: https://github.com/SwissDataScienceCenter/renku-python/blob/develop/renku/ui/service/serializers/common.py#L56
it never used/worked with ref.
so I merely cleaned it up without a breaking change, swagger was just wrong before